### PR TITLE
usm: tls: nodejs: Introduce NodeJS TLS support for USM

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

Introducing TLS monitoring on top of NodeJS applications with USM.
USM can now track HTTP, HTTP/2, and gRPC over openssl, istio, go-tls and node-js applications.

NodeJS is statically linked with openssl, thus our generic openssl support cannot work on NodeJS applications, we rely on dynamic linking for the process. In this PR we introduce a dedicate code path to identify NodeJS processes, and hook them as we do for openssl.

The new feature is similar to the implementation we have for istio, which means re-using the same hooks of openssl, but with a small change - a dedicate tag to identify the source of the traffic (openssl, istio, or nodejs).

### Motivation

Extend the TLS monitoring capabilities of USM to a new programming language.

### Additional Notes

The new feature can be enabled by:
```yaml
service_monitoring_config:
  enabled: true
  tls:
    nodejs:
      enabled: true
```

Or by setting the following environment variable `DD_SERVICE_MONITORING_CONFIG_TLS_NODEJS_ENABLED=true`

### CODE REVIEW

File: pkg/network/usm/monitor_tls_test.go [hunk 1]
```diff
@@ -780,3 +781,73 @@ func getHTTPLikeProtocolStats(monitor *Monitor, protocolType protocols.ProtocolT
...
+	require.NoError(t, err)
+
+	require.NoError(t, nodejs.RunServerNodeJS(t, key, cert, serverPort))
+	nodeJSPID, err := nodejs.GetNodeJSDockerPID()
+	require.NoError(t, err)
+
+	cfg := config.New()
+	cfg.EnableHTTPMonitoring = true
+	cfg.EnableNodeJSMonitoring = true
+	cfg.BPFDebug = true
```
Comment (Yumasi) [inline pkg/network/usm/monitor_tls_test.go:803]: I think we can disable eBPF debugging, like in our other tests.

**Review by Yumasi — CHANGES_REQUESTED** [review]

**Review by hmahmood — APPROVED** [review]

**Review by alai97 — APPROVED** [review]
Looks good, I opened https://github.com/DataDog/documentation/pull/22459 as a doc update, feel free to review and add suggestions!

File: pkg/network/usm/monitor_tls_test.go (hunk 1 above)
Comment (guyarb) [reply pkg/network/usm/monitor_tls_test.go:803]: oops, good catch

File: pkg/network/protocols/tls/nodejs/nodejs.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	_, err = io.Copy(destination, source)
+	if err != nil {
+		return err
+	}
+
+	err = destination.Sync()
+	if err != nil {
+		return err
+	}
+
```
[the remark this answers is not in the document]
Comment (guyarb) [reply pkg/network/protocols/tls/nodejs/nodejs.go:41]: removed

File: pkg/config/setup/system_probe.go [hunk 3]
```diff
@@ -230,6 +230,7 @@ func InitSystemProbeConfig(cfg pkgconfigmodel.Config) {
 	cfg.BindEnvAndSetDefault(join(smNS, "enable_http2_monitoring"), false)
 	cfg.BindEnvAndSetDefault(join(smNS, "enable_kafka_monitoring"), false)
 	cfg.BindEnvAndSetDefault(join(smNS, "tls", "istio", "enabled"), false)
+	cfg.BindEnv(join(smNS, "tls", "nodejs", "enabled"))
```
Comment (usamasaqib) [inline pkg/config/setup/system_probe.go:233]: No need to set a default value?

Comment (guyarb) [reply pkg/config/setup/system_probe.go:233]: it will be false by default

**Review by usamasaqib — APPROVED** [review]

**Review by Yumasi — APPROVED** [review]

**Review by richl-dd — APPROVED** [review]

**Pull request merged**